packages/@lightningjs/ui-components/src/mixins/withThemeStyles/utils.test.js

```diff
@@ -672,6 +672,39 @@ describe('generateComponentStyleSource', () => {
     });
   });
 
+  it('will properly incorporate custom modes and tones', () => {
+    const source = generateComponentStyleSource({
+      styleChain: [
+        {
+          style: {
+            base: {
+              color: 'primary'
+            }
+          }
+        },
+        {
+          style: {
+            selected: {
+              color: 'brand'
+            }
+          }
+        }
+      ]
+    });
+
+    expect(source).toStrictEqual({
+      unfocused_neutral: { color: 'primary', selected: { color: 'brand' } },
+      unfocused_inverse: { color: 'primary', selected: { color: 'brand' } },
+      unfocused_brand: { color: 'primary', selected: { color: 'brand' } },
+      focused_neutral: { color: 'primary', selected: { color: 'brand' } },
+      focused_inverse: { color: 'primary', selected: { color: 'brand' } },
+      focused_brand: { color: 'primary', selected: { color: 'brand' } },
+      disabled_neutral: { color: 'primary', selected: { color: 'brand' } },
+      disabled_inverse: { color: 'primary', selected: { color: 'brand' } },
+      disabled_brand: { color: 'primary', selected: { color: 'brand' } }
```

Comment:
I'm a little confused by this test and the result having "selected" nested inside of the default mode/tones. This test reads more like just adding a custom property that is an object versus adding a mode or tone of "selected." Is that the expectation?